### ISSUE TITLE
Rule request: strict-number-check

### ISSUE THREAD

**Issue by Shinigami92** [palantir/tslint#3890]
#### TypeScript code being linted

```ts
const value: number | null = 0;
// value is 0 and js sees 0 == false
if (value) {
	// ...
}
```

(Lint only if there is no other expression in the if)

#### Expected behavior

If `value` can also be `number`, then **warn** and suggest:
```ts
const value: number | null = 0;
if (value || value === 0) {
	// ...
}
```

**Comment by JoshuaKGoldberg** [comment]
@Shinigami92 does the `strict-boolean-expressions` satisfy your use case here? Seems pretty similar.

**Comment by Shinigami92** [comment]
@JoshuaKGoldberg I think yes
